compiler/resolution/functionResolution.cpp

```diff
@@ -3128,6 +3128,20 @@ static void adjustClassCastCall(CallExpr* call)
   }
 }
 
+static bool
+isDowncastToGenericClass(CallExpr* call, Type* targetType,
```

Comment:
In this function we can and do already assume the call is a cast, and don't use `call`. I think it would be more accurate to name it something like `isGenericSubclass` since I think that's what's being checked here, and remove the `call` parameter.